test/base/test_buffer.py

```diff
@@ -278,7 +280,60 @@ def test_pickle():
                        pbuf.weight[np.arange(len(pbuf))])
 
 
+def test_hdf5():
+    size = 100
+    buffers = {
+        "array": ReplayBuffer(size, stack_num=2),
+        "list": ListReplayBuffer(),
+        "prioritized": PrioritizedReplayBuffer(size, 0.6, 0.4)
+    }
+    buffer_types = {k: b.__class__ for k, b in buffers.items()}
+    device = 'cuda' if torch.cuda.is_available() else 'cpu'
+    rew = torch.tensor([1.]).to(device)
+    for i in range(4):
+        kwargs = {
+            'obs': Batch(index=np.array([i])),
+            'act': i,
+            'rew': rew,
+            'done': 0,
+            'info': {"number": {"n": i}, 'extra': None},
```

Comment:
`None` is very common in gym's `info`, however, the current implementation does not support it.

```
step0: info: {}
step1: info: {}
step2: info: {'timelimit.policy': 1}
```

Reply:
Good point. I suggest that the the converter first tries to write numpy arrays to HDF5 directly and, if that files, falls back to pickling the array instead. This added a bit of `try`-`except` code that maybe looks a bit confusing. Let me know what you think.